File: simulators/de.unistuttgart.iste.sqa.mpw.hamstersimulator.java/examples/src/main/java/de/unistuttgart/hamster/examples/Example01.java
package de.unistuttgart.hamster.examples;

import de.unistuttgart.hamster.main.SimpleHamsterGame;

public class Example01 extends SimpleHamsterGame {
    public static void main(final String[] args) {
        createInstance(Example01.class);
    }

    /**
     * How often Paule should repeat its behaviour.
     */
    private static final int SCENARIO_REPETITION_COUNT = 20;

    @Override
    protected void run() {
        displayInNewGameWindow();
        for (int i = 0; i < SCENARIO_REPETITION_COUNT; i++) {
            paule.putBreadcrumb();
            paule.move();
            paule.putBreadcrumb();
            paule.move();
            paule.pickGrain();
            paule.pickGrain();
            paule.putGrain();
            paule.putGrain();
            paule.turnLeft();
            paule.turnLeft();
            paule.move();
            paule.pickBreadcrumb();
            paule.move();
            paule.pickBreadcrumb();
            paule.turnLeft();
            paule.turnLeft();
        }
    }

}

File: examples/Example01.cpp
#include "SimpleHamsterGame.h"

using namespace hamster;

class Example01 : public hamstersimulator::SimpleHamsterGame {
protected:
    void run() override;
};

void Example01::run() {
    paule.move();
    paule.move();
    paule.pickGrain();
    paule.turnLeft();
    paule.turnLeft();
    paule.move();
    paule.move();
    paule.putGrain();
    paule.turnLeft();
    paule.turnLeft();
    paule.move();
}

int main() {
    hamstersimulator::SimpleHamsterGame::createInstance<Example01>();
    return 0;
}

